koap/src/commonMain/kotlin/Message.kt

```diff
@@ -291,17 +293,21 @@ sealed class Message {
              * - `0` (register) adds the entry to the list, if not present;
              * - `1` (deregister) removes the entry from the list, if present.
              */
-            enum class Registration { Register, Deregister }
+            sealed class Registration {
+                object Register: Registration()
+                object Deregister: Registration()
+            }
 
             /**
              * Constructs an [Observe] to be included in a GET request.
              *
              * @see Registration
              */
+            @JsName("constructorWithRegistration")
```

Comment:
Rather than expose this constructor, can we take a similar approach as https://github.com/JuulLabs/koap/pull/35/files#r513669441?

Reply:
Again, I like treating it as though it was a function on the companion object for JS. In this case, `fromRegistration`, and calling it as `Observe.fromRegistration(registration)`

Reply:
Concur, will go with this naming approach.